server.py

```diff
@@ -1,3 +1,15 @@
+class User:
```

Comment:
1. I'm not sure if I would have used a class for this. But. This is also good
2. You are doing something very similar to dataclasses. Take a [look here](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass)

Reply:
I have an issue using dataclasses,
`@dataclass()`
`class User:`
`submissions: list() = field(default_factory=list)`
`solvedQuestions: set() = field(default_factory=set)`
when using the above code, I am getting an error.
> NameError:  name 'field' is not defined

I don't know how to give default values other than this.
When I used `submissions: list() = []`, I got an error telling me to use default_factory.

Reply:
do we really need to use class?

Reply:
I prefer using classes when the data structure is complex, it simplifies the program. And it makes it  more secure as we can restrict the access to attributes only through methods. In future, if we try to extend the user profile, this will come handy. I am trying to use dataclasses instead as suggested as they are better for storing data, but am having troubles in doing so. Still if you prefer some other way to store data, we might do it. But classes were the simplest I could think of.

Reply:
classes really are overkill at this point but I'm not complaining since it's *something* as opposed to nobody writing any code at all.

This can be done with something really simple like:

```python
correct_attempts = defaultdict(set)  # formerly userlist?

def check_function(qno, user):
    if attempt_is_correct:
        correct_attempts[user].add(qno)

def ranks():
    order = [(user, len(solved)) for user, solved in correct_attempts.items()]
    order.sort(key=lambda x: x[1], reverse=True)
    return order
```

Reply:
Are you suggesting to make different data structures for submissions and correct_attempts?
We stored both in the class which was later changed to dataclass in the latest commit.

Reply:
you could store it in any fashion you want. What I'm saying is that this requires a very simple data structure. Classes is too powerful a tool for this.

Here's what I ask when I want to know if I should use classes. In order:

1. can I inherit something?
2. can I subclass something?
3. does it make sense to carry around information as a unit?
4. can meta programming help here?

Reply:
Ok. So i will try to have a simpler structure. I get your point.